From JBeulich@novell.com  Mon Aug 17 12:30:34 2009
From: "Jan Beulich" <JBeulich@novell.com>
Date: Mon, 17 Aug 2009 05:54:31 -0600
Subject: Staging: hv: adjust Hyper-V Kconfig
To: "Greg Kroah-Hartman" <GregKH@novell.com>
Content-Disposition: inline

From: Jan Beulich <JBeulich@novell.com>

Hyper-V sub-components' options should all depend on the base option.

The default of these sub-component options is also more reasonably set
to that of the base option (since it makes little sense to enable the
base option without the sub-component ones).

Signed-off-by: Jan Beulich <jbeulich@novell.com>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/hv/Kconfig |   16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

--- a/drivers/staging/hv/Kconfig
+++ b/drivers/staging/hv/Kconfig
@@ -6,23 +6,27 @@ config HYPERV
 	  Select this option to run Linux as a Hyper-V client operating
 	  system.
 
+if HYPERV
+
 config HYPERV_STORAGE
 	tristate "Microsoft Hyper-V virtual storage driver"
-	depends on SCSI && m
-	default n
+	depends on SCSI
+	default HYPERV
 	help
 	 Select this option to enable the Hyper-V virtual storage driver.
 
 config HYPERV_BLOCK
 	tristate "Microsoft Hyper-V virtual block driver"
-	depends on BLOCK && SCSI && m
-	default n
+	depends on BLOCK && SCSI
+	default HYPERV
 	help
 	  Select this option to enable the Hyper-V virtual block driver.
 
 config HYPERV_NET
 	tristate "Microsoft Hyper-V virtual network driver"
-	depends on NET && m
-	default n
+	depends on NET
+	default HYPERV
 	help
 	  Select this option to enable the Hyper-V virtual network driver.
+
+endif
